### PR TITLE
bugfix: destroy `pool` before the daemon process exits.

### PR DESCRIPTION
**Pull request by membphis**
I hereby granted the copyright of the changes in this pull request
to the authors of this openresty project.

### CODE REVIEW

File: util/mirror-tarballs [hunk 1]
```diff
@@ -65,6 +65,11 @@ if [ "$answer" = "Y" ]; then
     echo "$info_txt applying the resolver_conf_parsing patch for nginx"
     patch -p1 < $root/patches/nginx-$main_ver-resolver_conf_parsing.patch || exit 1
     echo
+
+    echo "$info_txt applying the daemon-destroy-pool patch for nginx"
+    patch -p1 < $root/patches/nginx-$main_ver-daemon-destroy-pool.patch || exit 1
+    echo
+
```
Comment (agentzh) [inline util/mirror-tarballs]: Extra blank line here?

File: patches/nginx-1.13.6-daemon-destroy-pool.patch [hunk 2]
```diff
@@ -0,0 +1,12 @@
+diff --git a/src/os/unix/ngx_daemon.c b/src/os/unix/ngx_daemon.c
+index ab672110..4bf09e77 100644
+--- a/src/os/unix/ngx_daemon.c
++++ b/src/os/unix/ngx_daemon.c
+@@ -23,6 +23,7 @@ ngx_daemon(ngx_log_t *log)
+         break;
+ 
+     default:
++        ngx_destroy_pool(ngx_cycle->pool);
```
Comment (agentzh) [inline patches/nginx-1.13.6-daemon-destroy-pool.patch]: Should add a code comment here to make it clear that we just make it "ASAN clean".

**Comment by membphis** [comment]
@agentzh fixed right now, please take a look when you have time.

**Comment by agentzh** [comment]
Merged with minor fixes. Thanks.

**Pull request closed without merging**